searx/utils.py

```diff
@@ -384,10 +384,14 @@ def load_module(filename, module_dir):
 
 
 def new_hmac(secret_key, url):
+    try:
+        secret_key_bytes = bytes(secret_key, 'utf-8')
+    except TypeError:
```

Comment:
Do you mind adding additional check to see if in case of `TypeError` `secret_key` is indeed `bytes` type?

Reply:
added commit for test case and a check for `bytes` type but this will not for bytearray

https://stackoverflow.com/questions/34869889/what-is-the-proper-way-to-determine-if-an-object-is-a-bytes-like-object-in-pytho